### PR TITLE
Feature/#324 updated point system

### PR DESCRIPTION
**Pull request by oconn**
# PR for issue #324.

## Ready for a PR?
 - [x] I have updated the CHANGELOG with an entry including a description, a link to my profile, and a link to the issue ticket. This change is filed under an Enhancement or Bug Fix, is shown under the relevant release tag name, and is included in my PR branch. 
 
## Implementation Notes:

- While there are no breaking changes, you will need to refresh your local db if testing / dev-ing with these changes loaded for the first time.

## Breaking changes:

- None

### CODE REVIEW

File: src/wombats/constants.clj [hunk 1]
```diff
@@ -39,18 +39,52 @@
...
+   :stats/deaths-by-steel-barrier-collision 0
+   :stats/deaths-by-wombat-collision 0
+   :stats/deaths-by-zakano-collision 0
+   :stats/deaths-by-poison 0
+   :stats/wood-barrier-collisions 0
+   :stats/steel-barrier-collisions 0
+   :stats/zakano-collisions 0
+   :stats/wombat-collisions 0
+   :stats/frames-blinded 0
+   :stats/frames-played 0})
```
Comment (dehli) [inline src/wombats/constants.clj]: What's the difference between: `:stats/frame-number` and `:stats/frames-played`?

Comment (oconn) [inline src/wombats/constants.clj]: `:stats/frame-number` is a key used (datomic) to sync stats for users to support replay. "Get the frame & associated stats at time x" `:stats/frames-played` is used to track how much of the game you were in.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,15 +4,17 @@ CHANGELOG
 ## Develop
 **Enhancements**
 * AWS Lambda Mock
-	[Eric Rochester](https://github.com/erochest) #[329](https://github.com/willowtreeapps/wombats-api/issues/329)
+	[Eric Rochester][/erochest] #[329](https://github.com/willowtreeapps/wombats-api/issues/329)
```
Comment (dehli) [inline CHANGELOG.md]: ^ tab

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,15 +4,17 @@ CHANGELOG
...
+    [Matt O'Connell][/oconn] #[324](https://github.com/willowtreeapps/wombats-api/issues/324)
 
 **Bug Fixes**
 * None
 
 ## QA (3.21.2017)
 **Enhancements**
 * Added new simulator template
-	[Matt O'Connell](https://github.com/oconn) *No Issue*
+	[Matt O'Connell][/oconn] *No Issue*
```
Comment (dehli) [inline CHANGELOG.md]: ^ tab

File: src/wombats/constants.clj [hunk 4]
```diff
@@ -39,18 +39,52 @@
...
+   :wombat-hit-bonus 10
+   :zakano-hit-bonus 8
+   :steel-barrier-hit-bonus 2
+   :wood-barrier-hit-bonus 2
+   :wombat-destroyed-bonus 25
+   :zakano-destroyed-bonus 15
+   :wood-barrier-destroyed-bonus 3
+   :steel-barrier-destroyed-bonus 25
+   ;; In game parameters
+   :shot-distance 5})
```
Comment (dehli) [inline src/wombats/constants.clj]: Should we also store how far the wombat sees in here?

File: src/wombats/game/finalizers.clj [hunk 5]
```diff
@@ -46,6 +46,30 @@
...
     (assoc-in game-state [:game/frame :frame/arena] updated-arena)))
 
+(defn- update-player-stats
+  [game-state]
+  (let [wombats (filter (fn [cell]
+                          (= :wombat (get-in cell [:contents :type])))
+                        (flatten (get-in game-state [:game/frame :frame/arena])))]
+    (reduce (fn [game-state-acc wombat]
+              (let [wombat-uuid (get-in wombat [:contents :uuid])
+                    mmm-smoked-wombat (some #(= :smoke (:type %))
```
Comment (dehli) [inline src/wombats/game/finalizers.clj]: What does `mmm` stand for?

Comment (oconn) [inline src/wombats/game/finalizers.clj]: sure is

File: resources/datomic/schema.dtm [hunk 6]
```diff
@@ -522,34 +552,100 @@
   :db/ident       :stats/wood-barriers-destroyed
   :db/valueType   :db.type/long
   :db/cardinality :db.cardinality/one
-  :db/doc         "Number of wood-barriers destroyed"}
+  :db/doc         "Number of wood barriers destroyed"}
```
Comment (erochest) [inline resources/datomic/schema.dtm]: Is this for barriers they destroy by ramming through them? (I see `wood-barriers-shot` below, so just guessing .... )

Comment (oconn) [inline resources/datomic/schema.dtm]: So technically you can shoot a barrier multiple times before it's destroyed (another player could be shooting it from the other side) so those values would be different.

**Pull request merged**